packages/docs/docs/tutorial/part3.md

```diff
@@ -0,0 +1,125 @@
+      {'\n'}
+      <CharacterField fieldName="race" />
+      {'\n'}
+      <CharacterField fieldName="list of no more than three weapons" />
+      {'\n'}
+      <CharacterField fieldName="list of no more than two spells" />
+    </>
+  );
+}
+```
+
+The component will generate text like:
+
+```
+The following is a character profile for an RPG game in JSON format:
+name: Aria
+class: Ranger
+race: Elf.
+list of no more than three weapons: Long sword, Short bow, Dagger.
+list of no more than two spells: Firebolt, Mage Armor.
+```
+
+where each field is generated in parallel by a component called `<CharacterField>`.
+
+## Generating one field of the character sheet
+
+```tsx filename="packages/tutorial/src/part3.tsx"
+function CharacterField(props: { fieldName: string }) {
+  return (
+    <>
+      {props.fieldName}:{' '}
+      <ChatCompletion>
+        <SystemMessage>
+          The user is generating a character sheet for a fantasy role-playing game. Your job is to provide a single
+          value for one of the fields of the character sheet. Please only return the chosen value, and no other
+          conversational text.
+        </SystemMessage>
+        <UserMessage>Please generate the character sheet field: {props.fieldName}</UserMessage>
+      </ChatCompletion>
+    </>
+  );
+}
+```
+
+`<CharacterField>` is simply a use of `<ChatCompletion>` to invoke the LLM. The
+`<SystemMessage>` component contains the system prompt to instruct the LLM in general
+how to respond, and the `<UserMessage>` component contains the specific prompt for this
+field in the character sheet.
+
+## Formatting the output as JSON and YAML
+
+Now, we're going to use two new components -- `<JsonChatCompletion>` and `<YamlChatCompletion>` --
+to take the plain-text output of `<CharacterGenerator>` and format it as JSON and YAML.
+
+```tsx filename="packages/tutorial/src/part3.tsx"
+const app = (
+  <Inline>
+    <CharacterGenerator />
+    {'\n\n'}
+    The following is a JSON representation of this character profile:{'\n'}
+    {(conversation) => (
+      <JsonChatCompletion>
+        <UserMessage>{conversation}</UserMessage>
+      </JsonChatCompletion>
+    )}
+    {'\n\n'}
+    And here is a YAML representation of this character profile:{'\n'}
+    {(conversation) => (
+      <YamlChatCompletion>
+        <UserMessage>{conversation}</UserMessage>
+      </YamlChatCompletion>
+    )}
+  </Inline>
+);
+```
+
+For example, the `<JsonChatCompletion>` component will emit:
+
+```
+{
+    "name": "Aria",
+    "class": "Ranger",
+    "race": "Elf",
+    "weapons": [
+        "Long sword",
+        "Short bow",
+        "Dagger"
+    ],
+    "spells": [
+        "Firebolt",
+        "Mage Armor"
+    ]
+}
+```
+
+Internally, the `<JsonChatCompletion>` and `<YamlChatCompletion>` components prompt the
+LLM to take the input data, format it as JSON or YAML, and check that the resulting objects
+correctly parse in the target format.
```

Comment:
This actually isn't what the component was originally designed to do, and given the prompts that we're providing in [the component](https://github.com/fixie-ai/ai-jsx/blob/d7d91afdbd0b3bc5a20af367b82d6e9f2277d0d9/packages/ai-jsx/src/batteries/constrained-output.tsx), it feels incidental if it does happen to work now. :smile:

That said, the component you're describing here would be a great thing to have.